### ISSUE TITLE
undefined method `to_liquid' for Locomotive::Steam::ContentTypeField::SelectOption

### ISSUE THREAD

**Issue by burabure** [locomotivecms/steam#74]
This code in my template 
```rb
{% for category in contents.conseils_astuces.category_options %}
  <li data-uk-filter=""><a href="#!">{{ category }}</a></li>
{% endfor %}
```

causes this error:

```sh
[LiveEditing] undefined method `to_liquid' for #<Locomotive::Steam::ContentTypeField::SelectOption:0x007f12e0c9a6b8>
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/models/entity.rb:22:in `method_missing'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-liquid-4.0.0/lib/liquid/context.rb:190:in `find_variable'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-liquid-4.0.0/lib/liquid/variable_lookup.rb:34:in `evaluate'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-liquid-4.0.0/lib/liquid/context.rb:164:in `evaluate'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-liquid-4.0.0/lib/liquid/variable.rb:78:in `render'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-liquid-4.0.0/lib/liquid/block_body.rb:105:in `render_token'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-liquid-4.0.0/lib/liquid/block_body.rb:87:in `block in render'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-liquid-4.0.0/lib/liquid/block_body.rb:74:in `each'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-liquid-4.0.0/lib/liquid/block_body.rb:74:in `render'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-liquid-4.0.0/lib/liquid/tags/for.rb:115:in `block (2 levels) in render'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-liquid-4.0.0/lib/liquid/tags/for.rb:102:in `each'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-liquid-4.0.0/lib/liquid/tags/for.rb:102:in `each_with_index'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-liquid-4.0.0/lib/liquid/tags/for.rb:102:in `block in render'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-liquid-4.0.0/lib/liquid/context.rb:128:in `stack'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-liquid-4.0.0/lib/liquid/tags/for.rb:101:in `render'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-liquid-4.0.0/lib/liquid/block_body.rb:105:in `render_token'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-liquid-4.0.0/lib/liquid/block_body.rb:87:in `block in render'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-liquid-4.0.0/lib/liquid/block_body.rb:74:in `each'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-liquid-4.0.0/lib/liquid/block_body.rb:74:in `render'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-liquid-4.0.0/lib/liquid/block.rb:15:in `render'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-liquid-4.0.0/lib/liquid/tags/inherited_block.rb:71:in `block in render'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-liquid-4.0.0/lib/liquid/context.rb:128:in `stack'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-liquid-4.0.0/lib/liquid/tags/inherited_block.rb:64:in `render'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/liquid/tags/inherited_block.rb:42:in `render'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-liquid-4.0.0/lib/liquid/block_body.rb:105:in `render_token'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-liquid-4.0.0/lib/liquid/block_body.rb:87:in `block in render'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-liquid-4.0.0/lib/liquid/block_body.rb:74:in `each'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-liquid-4.0.0/lib/liquid/block_body.rb:74:in `render'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-liquid-4.0.0/lib/liquid/block.rb:15:in `render'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/liquid/tags/extends.rb:10:in `block in render'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-liquid-4.0.0/lib/liquid/context.rb:128:in `stack'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/liquid/tags/extends.rb:8:in `render'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-liquid-4.0.0/lib/liquid/block_body.rb:105:in `render_token'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-liquid-4.0.0/lib/liquid/block_body.rb:87:in `block in render'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-liquid-4.0.0/lib/liquid/block_body.rb:74:in `each'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-liquid-4.0.0/lib/liquid/block_body.rb:74:in `render'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-liquid-4.0.0/lib/liquid/template.rb:215:in `block in render'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-liquid-4.0.0/lib/liquid/template.rb:269:in `with_profiling'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-liquid-4.0.0/lib/liquid/template.rb:214:in `render'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/liquid/template.rb:17:in `render'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/middlewares/renderer.rb:40:in `parse_and_render_liquid'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/middlewares/renderer.rb:22:in `render_page'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/middlewares/renderer.rb:10:in `_call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/middlewares/thread_safe.rb:13:in `call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/middlewares/thread_safe.rb:23:in `next'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/middlewares/thread_safe.rb:18:in `call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/middlewares/thread_safe.rb:23:in `next'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/middlewares/thread_safe.rb:18:in `call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-3.0.1/lib/locomotive/steam/middlewares/cache.rb:19:in `call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-3.0.1/lib/locomotive/steam/middlewares/catch_error.rb:13:in `call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-3.0.1/lib/locomotive/steam/middlewares/page_editing.rb:14:in `call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-3.0.1/lib/locomotive/steam/middlewares/missing_translations.rb:13:in `block in call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-3.0.1/lib/locomotive/steam/middlewares/missing_translations.rb:24:in `find_and_persist_missing_translations'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-3.0.1/lib/locomotive/steam/middlewares/missing_translations.rb:12:in `call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-3.0.1/lib/locomotive/steam/middlewares/wysihtml_css.rb:12:in `call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/middlewares/thread_safe.rb:23:in `next'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/middlewares/thread_safe.rb:18:in `call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/middlewares/path.rb:13:in `call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/middlewares/thread_safe.rb:23:in `next'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/middlewares/thread_safe.rb:18:in `call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/middlewares/thread_safe.rb:23:in `next'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/middlewares/thread_safe.rb:18:in `call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/middlewares/thread_safe.rb:23:in `next'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/middlewares/locale.rb:24:in `block in _call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/i18n-0.7.0/lib/i18n.rb:257:in `with_locale'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/middlewares/locale.rb:23:in `_call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/middlewares/thread_safe.rb:13:in `call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/middlewares/thread_safe.rb:23:in `next'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/middlewares/thread_safe.rb:18:in `call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/middlewares/thread_safe.rb:23:in `next'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/middlewares/timezone.rb:16:in `block in _call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/activesupport-4.2.5.2/lib/active_support/core_ext/time/zones.rb:48:in `use_zone'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/middlewares/timezone.rb:15:in `_call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/middlewares/thread_safe.rb:13:in `call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/middlewares/robots.rb:15:in `call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/middlewares/thread_safe.rb:23:in `next'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/middlewares/thread_safe.rb:18:in `call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/middlewares/logging.rb:18:in `call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/middlewares/thread_safe.rb:23:in `next'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/middlewares/thread_safe.rb:18:in `call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/middlewares/default_env.rb:15:in `call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/dragonfly-1.0.12/lib/dragonfly/middleware.rb:14:in `call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms_steam-1.0.1/lib/locomotive/steam/middlewares/favicon.rb:15:in `call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/rack-rewrite-1.5.1/lib/rack/rewrite.rb:24:in `call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/rack-1.6.4/lib/rack/builder.rb:153:in `call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/actionpack-4.2.5.2/lib/action_dispatch/routing/mapper.rb:51:in `serve'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/actionpack-4.2.5.2/lib/action_dispatch/journey/router.rb:43:in `block in serve'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/actionpack-4.2.5.2/lib/action_dispatch/journey/router.rb:30:in `each'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/actionpack-4.2.5.2/lib/action_dispatch/journey/router.rb:30:in `serve'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/actionpack-4.2.5.2/lib/action_dispatch/routing/route_set.rb:815:in `call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/railties-4.2.5.2/lib/rails/engine.rb:518:in `call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/railties-4.2.5.2/lib/rails/railtie.rb:194:in `public_send'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/railties-4.2.5.2/lib/rails/railtie.rb:194:in `method_missing'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/actionpack-4.2.5.2/lib/action_dispatch/routing/mapper.rb:51:in `serve'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/actionpack-4.2.5.2/lib/action_dispatch/journey/router.rb:43:in `block in serve'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/actionpack-4.2.5.2/lib/action_dispatch/journey/router.rb:30:in `each'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/actionpack-4.2.5.2/lib/action_dispatch/journey/router.rb:30:in `serve'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/actionpack-4.2.5.2/lib/action_dispatch/routing/route_set.rb:815:in `call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/locomotivecms-3.0.1/lib/locomotive/middlewares/site.rb:26:in `call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/dragonfly-1.0.12/lib/dragonfly/middleware.rb:14:in `call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/warden-1.2.6/lib/warden/manager.rb:35:in `block in call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/warden-1.2.6/lib/warden/manager.rb:34:in `catch'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/warden-1.2.6/lib/warden/manager.rb:34:in `call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/rack-1.6.4/lib/rack/etag.rb:24:in `call'
    /home/patricio/.rvm/gems/ruby-2.1.3/gems/rack-1.6.4/lib/rack/conditiona
```

Looking at the class here 
https://github.com/locomotivecms/steam/blob/master/lib/locomotive/steam/entities/content_type_field.rb#L73 and after a little bit of debugging I think this case slipped by on the 2 -> 3 update since there's no to_liquid method defined for the class or any fallbacks.

**Comment by elsilentforce** [comment]
Hello there, I got the same error trying to iterate a select option field, here is my content type YML
```name: Conseils & Astuces
slug: conseils_astuces
description: A description of the content type for the editors
label_field_name: title
order_by: _position
order_direction: asc
public_submission_enabled: false
public_submission_account_emails: []
fields:
- title:
    label: Title
    type: string
    required: true
    hint: Explanatory text displayed in the back office
    localized: false
- article_info:
    label: Article info
    type: text
    required: false
    hint: Top line above the content area
    localized: false
- category:
    label: Category
    type: select
    required: false
    hint: Explanatory text displayed in the back office
    localized: false
    select_options:
      fr:
      - Optimiser sa retraite
      - Mes interlocuteurs
      - Ma carrière
      - Demander sa retraite
      - Décès
- featured_image:
    label: Featured image
    type: file
    required: false
    hint: Explanatory text displayed in the back office
    localized: false
- content:
    label: Content
    type: text
    required: false
    hint: Explanatory text displayed in the back office
    localized: false
- conclusion:
    label: Conclusion
    type: text
    required: false
    hint: Explanatory text displayed in the back office
    localized: false
```

**Comment by did** [comment]
hi @burabure @elsilentforce, could you both test with Steam 1.2.0.beta1 please? I remember fixing a bug like this a couple of months ago. Thanks for your help!

@burabure, the *_options liquid method should return an array of strings (https://github.com/locomotivecms/steam/blob/master/lib/locomotive/steam/liquid/drops/content_entry_collection.rb#L85).

**Comment by burabure** [comment]
@did This seems to be solved on engine 3.1/steam 1.1, thanks for your help!
